libraries/Network/src/NetworkUdp.cpp

```diff
@@ -292,18 +292,14 @@ void NetworkUDP::flush() {}
 
 int NetworkUDP::parsePacket() {
   if (rx_buffer) {
-    return 0;
+    if (rx_buffer->full()) { // packet was not read yet
+      return rx_buffer->available();
+    }
+    clear(); // discard the rest of the packet
   }
   struct sockaddr_storage si_other_storage;  // enough storage for v4 and v6
   socklen_t slen = sizeof(sockaddr_storage);
-  int len = 0;
-  if (ioctl(udp_server, FIONREAD, &len) == -1) {
```

Comment:
How come this check for the length can be removed? 
I do see it is later assigned via `recvfrom`, but it does allow for an early escape so no buffer needs to allocated if the length cannot be determined or is 0 (is length of 0 possible?)

Reply:
sorry. it was a merge error